#16 Add npm & travis badges to README

planktonic opened the pull request:
**WIP**

Adds npm package badge to stylelint-config-bitcrowd README

**TODO**
Add travis badge to the same README

Fixes #16

SonniesEdge commented:
* It's very well organised! It explains things very clearly from the ground up. 
* However, we need to consider who this is talking to. At the moment the new text read like a very basic primer in what CSS, Sass and frontend web design is, whereas previously it assumed that it was onboarding a frontend developer with some experience and giving them an overview of our naming conventions and methodologies used.
* I suggest splitting this into sections, each aimed at a particular audience. Keep the previous text as quick-intro for experienced people, use the new text for new juniors who don't know what CSS is.
* This is a public document, so the sections about our Dropbox library are incompatible with that. Either this needs to be private or those sections need to go.
  * Following from this, where is this conversation taking place? Are we talking to the public? Or new people starting at Bitcrowd?
* Bonus questions are useful if you're asking people to do research. Why are we asking these questions? Is it part of a curriculum? This again relates to who this document is talking to.

Text-specific comments are made inline.

SonniesEdge commented on README.md:
Suggest explaining that Sass is compiled to CSS via asset pipeline or another build tool.

SonniesEdge commented on README.md:
Not sure if this is a reference to Monty Python or The Wiz. Either way, I would suggest using something less dependent on a specific sense of humour, which will make it more cross-culturally readable.

SonniesEdge commented on README.md:
HTML is capitalised.

SonniesEdge commented on README.md:
Overly paternalistic. Reads as shouting. Explain to the reader *why* they shouldn't do this, rather than telling them that they shouldn't.

SonniesEdge commented on README.md:
Would be good to have HTML examples alongside this. Slightly abstract without it.

SonniesEdge commented on README.md:
Again, comes across as shouting. Explain how they are different.

SonniesEdge commented on README.md:
It would be good to explain (with an exmplae) of how many classes together are combined together in the HTML layer to produce styles. Anyone looking up OOBEMITSCSS would find examples revolving around using `include` and `extend`, which we do not tend to do.

SonniesEdge commented on README.md:
As much as a dislike Donald Trump, I'd suggest leaving this out.

SonniesEdge commented on README.md:
Can we leave out the references to suicide, please? And "crazy".

SonniesEdge commented on README.md:
* Not sure what "The Good Parts™" refers to.
* We need to define DRY.

SonniesEdge commented on README.md:
Define DRY and WET.

SonniesEdge commented on README.md:
"of course" presumes previous knowledge. Explain why these are nested but other things are not.

SonniesEdge commented on README.md:
This should probably come after the Nesting section on #L289, as knowledge of not using nested is presumed here.

SonniesEdge commented on README.md:
"Reviled" reads paternalistically. Let's avoid universal assumptions about these things. Better would be "We do not use @import in CSS as it makes a new blocking request..." and then follow with "In Sass it concatenates...".

SonniesEdge commented on README.md:
See note about where this conversation is taking place. Public or private? References to Bitcrowd library shouldn't be made in public setting.

SonniesEdge commented on README.md:
What browsers do we support as standard?

SonniesEdge commented on README.md:
Presume "were developed in public" should be "were developed in private".

SonniesEdge commented on README.md:
"I've seen" - doesn't match voice of rest of document.

SonniesEdge commented on README.md:
"such horrors" - better written as "such non-scalable techniques".

SonniesEdge commented on README.md:
"This feature/bug is called the Cascade" - suggest "This deliberate feature is called the Cascade".

SonniesEdge commented on README.md:
Capitalisation of "Sass".

SonniesEdge commented on README.md:
Define/link GZIP. If the reader needs CSS explaining, they definitely need this.

planktonic commented on README.md:
Yup, removing all/most/many exclamation points 👍 

An explanation of atomic css would be out of place here as we don’t use atomic css, but I’ll add a link.

planktonic commented on README.md:
Yup good call (I’m still trying to wipe the slang use of 'crazy' out of my vocab. It’s sticky)

planktonic commented on README.md:
Yes definitely out of order, I’ll think about which passage to move where…

planktonic commented on README.md:
Yes, damn security-through-obscurity dropbox links…

@xhr15: links to books in our dropbox library totally make sense for this document, but as charlie points out above, that’s a problem as this is a public document.

I see two options:
- make this repo private
- remove the links (maybe create a separate reading list repo for frontend stuff and make that private?)

planktonic commented on README.md:
We don’t have a standard set of browsers we support. Each client has different needs & wishes, & their projects have different requirements & target audiences. This is always done on a per-project basis. The sentence was largely meaningless as we *always* request & receive different requirements from clients, and that has never been 'only the newest'.

planktonic commented on README.md:
I’ve fixed the spelling & grammar (it was a horrible sentence), and then yes — cut the sentence into two. I’ve also removed the parenthetical as this doc is public.

But I definitely meant 'in public' — all these techniques get discussed and improved on a lot in the open

planktonic commented on README.md:
Yes both these statements are personal opinion, as is the concept of recommending further reading — for this (and the 'read more' sections) I think it’s totally acceptable and natural-feeling to slip into a more informal tone to help indicate this switch.

…but I’ve updated the statement to make it clearer the assessment of cheesiness level is also opinion, rather than stating it so flatly.

planktonic commented on README.md:
Sure, done

planktonic commented on README.md:
👍  I found/replaced to all mis-capitalisations of Sass & HTML

planktonic commented on README.md:
Good parts refers to a super famous JS book from o’reilly. It’s famous because douglas rockford (one of the developers of JS) wrote it, it arguably lives up to its name & the hype, and because of this image:
![javascript the good parts](http://sealedabstract.com/wp-content/uploads/2013/05/JavaScript-the-good-parts.jpg)

Not important if the connection is made really, but I’ll see if I can find a suitable link to the book to make this a link at least.

First mention of DRY is now linked to the wiki page

planktonic commented on README.md:
Definitions are really outside of the scope of this doc — first instance of DRY is now linked to wiki though

planktonic commented on README.md:
adding 'deliberate' seems superfluous (if a feature isn't deliberate it’s a bug, no?). But I’ll make it clear somehow that it’s a feature that causes problems/is avoided/etc (that was the goal of that clause)

planktonic commented on README.md:
No reference at all, though it’s possibly a cliché. It’s merely part of this intro/history, and flows into the next heading

planktonic commented on README.md:
I’m a little alarmed by 'paternalistic'. Do you mean this is sexist? (Or did you just conflate 'paternal' and 'patronising'?)

Regardless the *why* is already given, in the same paragraph.

planktonic commented on README.md:
Yup, I hoped the example showed that — each root element of a component has the base class *and* the modifier class (I never show any different in the examples, I hope!), hence no `@extend`.

Perhaps it’s not clear enough? Let me know, but regardless — this section could do with expanding beyond one line and an example.

planktonic commented on README.md:
All of this requires previous knowledge, I don’t feel we need to explain everything in depth — I’ve kept the line but asked the reader to consider *why*, in a parenthetical.

planktonic commented:
Ok this could get confusing. I’m replying directly to inline comments, this is a reply to the not-inline/first comment. Let’s see if we know where we are at the end of it…


**Target audience**

In my experience lots of frontend devs may identify with the statement they 'can do CSS'; this most often means they understand what classes are, how to make something appear further to right/in a different colour etc. but little about CSS 'architecture':
- organisation of files, classes & selectors in order to achieve an extendable, reliable, logical & small CSS codebase
- how more general programming principles (DRY, SRP…) are being applied to CSS, why & how that helps.

This could fit the devs at our last on-site client (for example), and also our juniors.

I selected the principles I included here because they fill those holes in devs’ knowledge, and because it’s such things I most often see done poorly.


**Organisation**

Splitting into sections would be ideal, moreso as we add content. I’ve added an issue to look at that (#20) independently of this PR.


**Bonus Questions**

I added bonus questions to give direction for further learning — what might be interesting for the reader to investigate, but is out of the scope of this doc/too much detail? Same motivation with the 'read more' sections, just a different approach that may grab different people more than a reading list. No curriculum implied outside this document, and I’m certainly not expecting answers :)

planktonic commented on README.md:
Also confused by this one — what do you mean by 'paternalistic'? Is this passage sexist?

planktonic commented on README.md:
I want this to be a chatty/personable intro to how/why people have developed the techniques this doc describes, not a dry statement of a series of facts; I used the word 'horror' here for some minor amount of colour/humour.

The problems with the techniques I’ve listed here as 'horrors' are manifold, and don't include scaleability — for example table-based layouts scale just fine, it’s the lack of semantics/accessibility (& later on responsiveness) that’s the problem.

planktonic commented on README.md:
Introduces OOBEMITSCSS before ITCSS

planktonic commented:
Ok updated, edited & expanded. Please all take another look if & when you have a moment!

planktonic committed:
Adds npm package badge to stylelint-config-bitcrowd README
